### DISCUSSION TITLE
Derive GitHub Actions digest from release git tag?

### DISCUSSION THREAD

**Discussion by noahnu** [renovatebot/renovate#23136] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Using the `github-actions` manager, in following with GitHub's security hardening guidelines for [using third-party actions](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions), I am specifying git shas for my 3rd party actions.

When Renovate creates pull requests to update these actions, it us updating the digest to the _latest_ sha on the default branch of the action's repository. This presents two issues:

- Not every commit in an action's repository is necessarily a functional commit. It's not necessarily a common pattern, but see the action [slackapi/slack-github-action](https://github.com/slackapi/slack-github-action) which only creates a "dist" directory on the release tags (similar to the gh-pages pattern, it overwrites the branch's contents).
- When Renovate puts up a pull request updating the digest, I have no way to know whether it's a breaking change or not without navigating to the action's repository and doing a diff of the shas.
  - Renovate doesn't show changelogs when using digests.
  - On digest update PRs, the package name isn't even linkable, so I really have to open a new browser/tab and search for the changes myself:
  
    ![image](https://github.com/renovatebot/renovate/assets/1297096/1a0108e9-704e-469b-b23e-7bc141c5acbe)

Instead, I want Renovate to operate on release git tags (e.g. v1.0.0, v1.2.3) and then when it goes to apply the change, I want it to translate the tag to a digest. Essentially, I want the security hardening of using a digest (where it's unlikely to change via a malicious actor) with the semantic meaning, readability, and changelogs offered by versioning.

Is this something already supported?

### What you tried so far.

I tried digging through the config docs, but the closest I found was some configs around extracting versioning and adjusting PR title/description settings when using digests.

### Relevant debug logs

No logs.

**Comment by noahnu** [comment, marked as answer]
Ah I missed this. Looks like we can use a comment:

https://docs.renovatebot.com/modules/manager/github-actions/#additional-information

```
name: build

on: [push]

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@af513c7a016048ae468971c52ed77d9562c7c819 # renovate: tag=v1.0.0
```

**Comment by viceice** [comment]
add the git tag behind the git Sha with a comment, like `# 1.2.3`.

same syntax is supported by renovate and dependabot
